### DISCUSSION TITLE
Confusion about Golang packages

### DISCUSSION THREAD

**Discussion by gschleic** [vercel/vercel#5725] [Help]
I am able to successfully run a project with the following general "structure" using `vercel dev` but it fails when I attempt to deploy it using `vercel` :

**root/go.mod**
```
package "mydomain.com/mymodule"
```

**root/api/handler.go** 
```
import "mydomain.com/mymodule/otherpackage"
result:= otherpackage.ExportedFn()
```
**root/otherpackage/otherpackage.go** 
```
package otherpackage

function ExportedFn( ... ) 

```

It seems like the Vercel go runtime is trying to resolve "myydomain.com/otherpackage" as a public package
and I am getting the following error:
```
/handler imports
        mydomain.com/mymodule/otherpackage: cannot find module providing package  mydomain.com/mymodule/otherpackage
```

I tried looking online for guidance but a lot of the material around go serverless seems out of date or shows a simple handler OR handler which only requires external public packages. I suspect a runtime issue with GOROOT or GOPATH? Any ideas?

I looked at the advanced usage section of the runtime where it talks about setting a GIT_CREDENTIALS token to use "private"
packages, but the private package above is not in git (it sits at a peer level with the Vercel root "api/" directory with a monorpo) . Is this 
the right file structure?

**Comment by Kawemi** [comment]
I have the same problem. I thought I could put my `go.mod` insde the `/api` folder but nothing seems to work. Did you find something that worked for your setup?

**Comment by shuiRong** [comment]
same problem here.

(here is a [sample](https://github.com/WeiboTrendingBackup/WeiboTrendingAPI/tree/75c64ae378ec0f3b31d2721f36e13fb3f9a58282) that could reproduce it, if there is anyone could provide help. thanks, advanced.)

**Comment by codeallthethingz** [comment]
I have the same problem.  My package used to work with the `go.mod` in the /api folder.  But since I upgraded to Macos Big Sur, I'm starting to get strange errors

```
go: cannot find main module, but found .git/config in /Users/will/workspace/org-chart
	to create a module there, run:
	cd ../../../.. && go mod init
Error! Failed to start dev server for "api/index.go" (code=1, signal=null)
```

**Reply by shuiRong** [reply]
I solved this problem by typing `go mod init xxxxx` in the root path of the project.

ref: https://github.com/xuthus5/zeit-serverless/issues/1

**Comment by igoel** [comment]
Still same problem here, someone found a solution?

**Comment by myanpetra99** [comment]
same here, still haven't found any solutions?
